### PR TITLE
JCLOUDS-1167 oneandone-privatenetwork-api

### PR DESCRIPTION
**Pull request by alibazlamit**
@nacx Ready for review thanks.

### CODE REVIEW

File: docker/src/test/java/org/jclouds/docker/compute/strategy/PredicateLocateImageByNameTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+   public void testRepoTagLatestWithHost() {
+      final Predicate<Image> predicate = DockerComputeServiceAdapter
+            .createPredicateMatchingRepoTags("docker.io/kwart/alpine-ext");
+      assertFalse(predicate.apply(IMAGE_REPO_TAGS_MULTI));
+      assertFalse(predicate.apply(IMAGE_REPO_TAGS_EMPTY));
+      assertTrue(predicate.apply(IMAGE_REPO_TAGS_WITH_HOST));
+   }
+
+}
```
Comment (nacx) [inline docker/src/test/java/org/jclouds/docker/compute/strategy/PredicateLocateImageByNameTest.java]: Make sure your branch is properly rebased. This class shouldn't be part of the PR.

File: oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/GenericRequest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.oneandone.rest.domain;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class GenericRequest {
```
Comment (nacx) [inline oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/GenericRequest.java]: This class seems to not be used at all. Remove it.

File: oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/PrivateNetwork.java [hunk 3]
```diff
@@ -58,4 +58,120 @@
...
+      }
+
+      @AutoValue
+      public abstract static class CreateServer {
+
+         public abstract List<String> servers();
+
+         @SerializedNames({"servers"})
+         public static CreateServer create(List<String> servers) {
+            return new AutoValue_PrivateNetwork_Server_CreateServer(servers == null ? ImmutableList.<String>of() : servers);
```
Comment (nacx) [inline oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/PrivateNetwork.java]: Enforce an immutable list when present too.

File: oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/PrivateNetworksApiLiveTest.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+   }
+
+   @Test
+   public void testList() {
+      privateNetworks = privateNetworkApi().list();
+
+      Assert.assertTrue(privateNetworks.size() > 0);
+   }
+
+   @Test(dependsOnMethods = "testList")
```
Comment (nacx) [inline oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/PrivateNetworksApiLiveTest.java]: There is no need for this dependency.

File: oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/PrivateNetworksApiLiveTest.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+   public void testListWithOption() {
+      GenericQueryOptions options = new GenericQueryOptions();
+      options.options(0, 0, null, "jclouds", null);
+      List<PrivateNetwork> resultWithQuery = privateNetworkApi().list(options);
+
+      Assert.assertTrue(resultWithQuery.size() > 0);
+   }
+
+   @Test(dependsOnMethods = "testListWithOption")
```
Comment (nacx) [inline oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/PrivateNetworksApiLiveTest.java]: No need for the dependency.

File: oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/PrivateNetworksApiLiveTest.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+   @Test(dependsOnMethods = "testAttachServer")
+   public void testListServers() {
+      assertPrivateNetworkAvailable(ServerPrivateNetworkRef.create(currentServer.id(), currentPrivateNetwork.id()));
+      List<PrivateNetwork.Server> servers = privateNetworkApi().listServers(currentPrivateNetwork.id());
+
+      Assert.assertTrue(servers.size() > 0);
+   }
+
+   @Test(dependsOnMethods = "testListServers")
```
Comment (nacx) [inline oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/PrivateNetworksApiLiveTest.java]: Better depend on the attach test too.

File: pom.xml [hunk 7]
```diff
@@ -37,7 +37,7 @@
   </scm>
 
   <properties>
-    <jclouds.version>2.0.0-SNAPSHOT</jclouds.version>
+    <jclouds.version>2.0.0-SNAPSHOT</jclouds.version>
```
Comment (nacx) [inline pom.xml]: Discard this change

**Comment by nacx** [comment]
Thanks @alibazlamit! Take care of reformatting the code to use 3 space indentation.

**Comment by alibazlamit** [comment]
@nacx  Changed applied and re based correctly

**Comment by nacx** [comment]
Pushed to master as [80e5c911](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/80e5c911). Thanks @alibazlamit!
